crates/iota-indexer/src/models/objects.rs

```diff
@@ -160,6 +180,94 @@ pub struct StoredHistoryObject {
     pub df_kind: Option<i16>,
 }
 
+impl StoredHistoryObject {
+    pub async fn try_into_past_object_read(
+        &self,
+        package_resolver: Arc<Resolver<impl PackageStore>>,
+    ) -> Result<PastObjectRead, IndexerError> {
+        let object_ref = self.get_object_ref()?;
+
+        let object: Object = self.clone().try_into()?;
+
+        if !object.digest().is_alive() {
+            return Ok(PastObjectRead::ObjectDeleted(object_ref));
+        }
+
+        let Some(move_object) = object.data.try_as_move().cloned() else {
+            return Ok(PastObjectRead::VersionFound(object_ref, object, None));
+        };
+
+        let move_type_layout = package_resolver
+            .type_layout(move_object.type_().clone().into())
+            .await
+            .map_err(|e| {
+                IndexerError::ResolveMoveStruct(format!(
+                    "failed to convert into object read for obj {}:{}, type: {}. error: {e}",
+                    object.id(),
+                    object.version(),
+                    move_object.type_(),
+                ))
+            })?;
+
+        let move_struct_layout = match move_type_layout {
+            MoveTypeLayout::Struct(s) => Ok(s),
+            _ => Err(IndexerError::ResolveMoveStruct(
+                "MoveTypeLayout is not Struct".to_string(),
+            )),
+        }?;
+
+        Ok(PastObjectRead::VersionFound(
+            object_ref,
+            object,
+            Some(*move_struct_layout),
+        ))
+    }
+
+    pub fn get_object_ref(&self) -> Result<ObjectRef, IndexerError> {
+        let object_id = ObjectID::from_bytes(self.object_id.clone()).map_err(|_| {
+            IndexerError::Serde(format!("Can't convert {:?} to object_id", self.object_id))
+        })?;
+        if let Some(object_digest) = &self.object_digest {
```

Comment:
Would it make sense to write here sth like:
```
let object_digest = self.object_digest.ok_or(IndexerError::Serde(...))?;
```
?

We could get rid of a lot of indentation that way.